src/DIPS.Xamarin.UI/Internal/Xaml/SheetView.xaml.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using DIPS.Xamarin.UI.Controls.Sheet;
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+
+namespace DIPS.Xamarin.UI.Internal.xaml
+{
+    /// <summary>
+    /// A sheetview that is used inside of a <see cref="SheetBehavior"/>
+    /// </summary>
+    /// <remarks>This is a internal Xaml control that should only be used in a <see cref="SheetBehavior"/></remarks>
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    public partial class SheetView : ContentView
+    {
+        private readonly SheetBehavior m_sheetBehaviour;
+        /// <summary>
+        /// Constructs a <see cref="SheetView"/>
+        /// </summary>
+        /// <param name="sheetBehavior"></param>
+        public SheetView(SheetBehavior sheetBehavior)
+        {
+            InitializeComponent();
+            OuterSheetFrame.BindingContext = m_sheetBehaviour = sheetBehavior;
+        }
+
+        /// <summary>
+        /// The height that the sheet content needs if it should display all of its content
+        /// </summary>
+        internal double SheetContentHeighRequest =>
+            SheetContent.Content.Height + HandleBoxView.Height + OuterSheetFrame.Padding.Top + OuterSheetFrame.Padding.Bottom + OuterSheetFrame.CornerRadius;
+
+        /// <summary>
+        /// The internal outer sheet frame of the view
+        /// </summary>
+        internal Frame SheetFrame => OuterSheetFrame;
+
+        private double m_newY;
+        private void OnDrag(object sender, PanUpdatedEventArgs e)
+        {
+            if (!m_sheetBehaviour.IsDraggable) return;
+            if (m_newY == 0) m_newY = SheetFrame.TranslationY;
+
+            switch (e.StatusType)
+            {
+                case GestureStatus.Started:
+                    m_sheetBehaviour.ToggleIsDragging();

```

Comment:
m_sheetBehaviour.ToggleIsDragging is only shifting a bool. If you drag with 2 fingers you might get it out of sync while draggin?

Reply:
Fixed, I removed the `ToggleIsDragging` and made a internal property instead (I didnt know that I could create internal properties 👎 ), which made it more readable as well.